Makefile

```diff
@@ -16,97 +16,103 @@ METADATA_DEFAULTS_FILE=${MAKEFILE_PATH}/pkg/config/defaults/aemm-metadata-defaul
 
-fmt:
-	goimports -w ./
-
 compile:
 	@echo ${MAKEFILE_PATH}
 	go build -a -tags aemm${GOOS} -ldflags '-X "${DEFAULT_VALUES_VAR}=${ENCODED_METADATA_DEFAULTS}"' -o ${BUILD_DIR_PATH}/${BINARY_NAME} ${MAKEFILE_PATH}/cmd/amazon-ec2-metadata-mock.go
 
-build: create-build-dir validate-json compile
+validate-json:
+	${MAKEFILE_PATH}/test/json-validator
 
-build-and-test: build test
+build: create-build-dir validate-json compile
 
-build-binaries:
-	${MAKEFILE_PATH}/scripts/build-binaries -d -p ${SUPPORTED_PLATFORMS} -v ${VERSION}
+unit-test: create-build-dir
+	go test ${MAKEFILE_PATH}/... -v -coverprofile=coverage.txt -covermode=atomic -outputdir=${BUILD_DIR_PATH}
 
-docker-build:
-	${MAKEFILE_PATH}/scripts/build-docker-images -d -p ${GOOS}/${GOARCH} -r ${IMG} -v ${VERSION}
+validate-readme:
+	${MAKEFILE_PATH}/test/readme-validator
 
-docker-run:
-	docker run ${IMG_W_TAG}
+e2e-test: build
+	${MAKEFILE_PATH}/test/e2e/run-tests
 
-docker-push:
-	@echo ${DOCKERHUB_TOKEN} | docker login -u ${DOCKERHUB_USERNAME} --password-stdin
-	docker push ${IMG_W_TAG}
+helm-lint-test:
+	${MAKEFILE_PATH}/test/helm/chart-test.sh -l
 
-build-docker-images:
-	${MAKEFILE_PATH}/scripts/build-docker-images -d -p ${SUPPORTED_PLATFORMS} -r ${IMG} -v ${VERSION}
+helm-app-version-test:
+	${MAKEFILE_PATH}/test/helm/helm-app-version-test.sh
 
-push-docker-images:
-	@echo ${DOCKERHUB_TOKEN} | docker login -u ${DOCKERHUB_USERNAME} --password-stdin
-	${MAKEFILE_PATH}/scripts/push-docker-images -p ${SUPPORTED_PLATFORMS} -r ${IMG} -v ${VERSION} -m
+helm-e2e-test:
+	${MAKEFILE_PATH}/test/helm/chart-test.sh
 
-version:
-	@echo ${VERSION}
+license-test:
+	${MAKEFILE_PATH}/test/license-test/run-license-test.sh
 
-latest-tag:
-	@echo ${LATEST_TAG}
+go-report-card-test:
+	${MAKEFILE_PATH}/test/go-report-card-test/run-report-card-test.sh
 
-image:
-	@echo ${IMG_W_TAG}
+test: unit-test e2e-test helm-app-version-test helm-e2e-test license-test go-report-card-test
 
-upload-resources-to-github:
-	${MAKEFILE_PATH}/scripts/upload-resources-to-github
+build-binaries:
+	${MAKEFILE_PATH}/scripts/build-binaries -d -p ${SUPPORTED_PLATFORMS} -v ${VERSION}
 
 generate-k8s-yaml:
 	${MAKEFILE_PATH}/scripts/generate-k8s-yaml
 
-sync-readme-to-dockerhub:
-	${MAKEFILE_PATH}/scripts/sync-readme-to-dockerhub
-
-unit-test: create-build-dir
-	go test ${MAKEFILE_PATH}/... -v -coverprofile=coverage.txt -covermode=atomic -outputdir=${BUILD_DIR_PATH}
-
-e2e-test: build
-	${MAKEFILE_PATH}/test/e2e/run-tests
+gen-helm-chart-archives:
+	${MAKEFILE_PATH}/scripts/generate-helm-chart-archives
 
-validate-json:
-	${MAKEFILE_PATH}/test/json-validator
+upload-resources-to-github:
+	${MAKEFILE_PATH}/scripts/upload-resources-to-github
 
-validate-readme:
-	${MAKEFILE_PATH}/test/readme-validator
+build-release-assets: create-build-dir build-binaries generate-k8s-yaml gen-helm-chart-archives
 
-helm-lint-test:
-	${MAKEFILE_PATH}/test/helm/chart-test.sh -l
+release: build-release-assets upload-resources-to-github
```

Comment:
release should also `release-docker`

Reply:
Hm.. so then remove the Docker Deploy stage from .travis.yml? Still executes sequentially, but a little more readable (when separate) in Travis

Reply:
you can modify travis to use the individual release target, but I think we need to have an aggregate that we can run locally if necessary that does everything.